cellprofiler/modules/threshold.py

```diff
@@ -932,6 +932,10 @@ def get_local_threshold(self, image, mask, volumetric):
         elif numpy.all(image_data == image_data[0]):
             local_threshold = numpy.full_like(image_data, image_data[0])
 
+        elif numpy.unique(numpy.where(numpy.isnan(image_data),0,image_data)).shape == (1,):
```

Comment:
`np.all((x == 0) | np.isnan(x))` avoids writing a new `image_data`-sized array/matrix.

Reply:
Or at least it writes 2 x `image_data` shaped arrays, but the dtype is `bool`

Reply:
See, I knew you'd have a cleaner idea